mod_reforged/hooks/items/weapons/weapon.nut

```diff
@@ -71,6 +71,23 @@
 		}
 	}
 
+	q.isDroppedAsLoot = @(__original) function()
+	{
+		if (!this.item.isDroppedAsLoot())
+		{
+			return false;
+		}
+
+		local actor = this.getContainer().getActor();
+		// Same condition as vanilla has for isPlayer
+		if (this.m.LastEquippedByFaction == this.Const.Faction.Player || this.getContainer() != null && !::MSU.isNull(actor) && this.isKindOf(actor.get(), "player"))
+		{
+			return true;	// Player Weapons now always drop guranteed
+		}
```

Comment:
You need to first check if container is null. Currently you check for that condition in the next `if` block but you already use `getActor` here assuming the container is not null. Also `null` check on actor is unnecessary as `isKindOf` will already return false if that is the case. So we can optimize this a bit.

```suggestion
		if (this.m.LastEquippedByFaction == ::Const.Faction.Player || !::MSU.isNull(this.getContainer()) && ::MSU.isKindOf(this.getContainer().getActor(), "player"))
		{
			return true;
		}
```